### PR TITLE
Doc wallet_scanQRCode

### PR DESCRIPTION
**Pull request by alexandratran**
Add wallet_scanQRCode specs (based off of the [manual reference](https://docs.metamask.io/wallet/reference/rpc-api/#wallet_scanqrcode) in the docs).

### CODE REVIEW

**Review by vandan — APPROVED** [review]
This is a helpful addition.
@shanejonas please review for technical accuracy.

File: openrpc.json [hunk 1]
```diff
@@ -511,6 +511,36 @@
         }
       ]
     },
+    {
+      "name": "wallet_scanQRCode",
+      "tags": [
+        {
+          "$ref": "#/components/tags/MetaMask"
```
Comment (shanejonas) [inline openrpc.json:518]: I think we are using `MetaMask` as a tag for extension and `Mobile` for mobile. we should think through the tags here

File: openrpc.json [hunk 2]
```diff
@@ -511,6 +511,36 @@
...
+        },
+        {
+          "$ref": "#/components/tags/Mobile"
+        }
+      ],
+      "summary": "Requests that the user scan a QR code.",
+      "description": "Requests that the user scan a QR code using their device camera. Introduced by [EIP-945](https://github.com/ethereum/EIPs/issues/945).",
+      "params": [
+        {
+          "name": "regex",
```
Comment (BelfordZ) [inline openrpc.json:528]: Is it a set of regex(S) or is it expecting a single regex?

ie, does the method call look like:

```
method: 'wallet_scanQRCode',
params: [ [ 'regexGoesHere', 'anotherRegexGoesHere'  ]  ]
```

or 

```
method: 'wallet_scanQRCode',
params: [ 'anotherRegexGoesHere'  ]
```


or 

```
method: 'wallet_scanQRCode',
params: [ [ 'anotherRegexGoesHere' ]  ]
```

File: openrpc.json [hunk 3]
```diff
@@ -511,6 +511,36 @@
...
+          "schema": {
+            "type": "array"
+          }
+        }
+      ],
+      "result": {
+        "name": "ScanQRCodeResult",
+        "description": "A string corresponding to the scanned QR code. If a regex string is provided, the resulting string matches it. If no regex string is provided, the resulting string matches an Ethereum address. If neither condition is met, the method returns an error.",
+        "schema": {
+          "type": "string"
```
Comment (BelfordZ) [inline openrpc.json:540]: schema should have a `title`

File: openrpc.json (hunk 2 above)
Comment (alexandratran) [reply openrpc.json:528]: I believe #2 according to the manual reference: https://docs.metamask.io/wallet/reference/rpc-api/#wallet_scanqrcode

**Pull request merged**